source/speech.py

```diff
@@ -937,26 +937,19 @@ def getSpeechTextForProperties(reason=controlTypes.REASON_QUERY,**propertyValues
 		textList.append(value)
 	states=propertyValues.get('states')
 	realStates=propertyValues.get('_states',states)
-	if states is not None:
-		positiveStates=controlTypes.processPositiveStates(role,realStates,reason,states)
-		textList.extend([controlTypes.stateLabels[x] for x in positiveStates])
-	if 'negativeStates' in propertyValues:
-		negativeStates=propertyValues['negativeStates']
-	else:
-		negativeStates=None
-	if negativeStates is not None or (reason != controlTypes.REASON_CHANGE and states is not None):

```

Comment:
> Could you clarify this line? Was there a particular reason why if negativeStates is not None wasn't enough?

The `negativeStates` argument is only provided for `REASON_CHANGE`. It specifies states that have just changed from positive to negative. However, for other reasons (e.g. `REASON_FOCUS`), we still want to report certain states if they aren't present; e.g. "not checked".